packages/gitbook/src/components/SitePage/fetch.ts

```diff
@@ -106,6 +119,50 @@ async function resolvePage(context: GitBookSiteContext, params: PagePathParams |
     return undefined;
 }
 
+async function resolvePageMetaLinks(context: GitBookSiteContext, pageId: string) {
+    const pageMetaLinks = await getDataOrNull(
+        context.dataFetcher.listRevisionPageMetaLinks({
+            spaceId: context.space.id,
+            revisionId: context.revisionId,
+            pageId,
+        })
+    );
+
+    if (pageMetaLinks) {
+        const spacesById = context.siteSpaces.reduce(
+            (acc, { space }) => {
+                acc[space.id] = space;
+                return acc;
+            },
+            {} as Record<string, Space>
+        );
+
+        const [resolvedCanonical, ...resolvedAlternates] = await Promise.all(
+            [pageMetaLinks.canonical, ...(pageMetaLinks.alternates || [])]
+                .filter(filterOutNullable)
```

Comment:
Your spread logic line 140 doesn't work if you do a filter, as the canonical is optional, it cam mean an alternate will be skipped and shown as canonical